### ISSUE TITLE
Installation with pyenv/virtualenvs?

### ISSUE THREAD

**Issue by BirdExpert123** [vapoursynth/vapoursynth#855]
Hi, I managed to get vapoursynth running using my system's python3 version (3.8.10) without any issues at all but I was wondering if there was anyway that I could install it using a virtual environmnt instead?

So far what I did was:
1. `pyenv virtualenv 3.9.9 encoding && pyenv activate encoding`
2. `git clone https://github.com/vapoursynth/vapoursynth  && cd vapoursynth`
3. `./autogen.sh && ./configure && make && sudo make install`

However when I run `vspipe --version` I get the following:
```Could not find platform independent libraries <prefix>
Could not find platform dependent libraries <exec_prefix>
Consider setting $PYTHONHOME to <prefix>[:<exec_prefix>]
Python path configuration:
  PYTHONHOME = (not set)
  PYTHONPATH = '/home/me/.pyenv/versions/encoding/lib/python3.9/site-packages'
  program name = 'python3'
  isolated = 0
  environment = 1
  user site = 1
  import site = 1
  sys._base_executable = '/home/me/.pyenv/shims/python3'
  sys.base_prefix = '/usr'
  sys.base_exec_prefix = '/usr'
  sys.platlibdir = 'lib'
  sys.executable = '/home/me/.pyenv/shims/python3'
  sys.prefix = '/usr'
  sys.exec_prefix = '/usr'
  sys.path = [
    '/home/me/.pyenv/versions/encoding/lib/python3.9/site-packages',
    '/usr/lib/python39.zip',
    '/usr/lib/python3.9',
    '/usr/lib/lib-dynload',
  ]
Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
Python runtime state: core initialized
ModuleNotFoundError: No module named 'encodings'

Current thread 0x00007f25f1f30740 (most recent call first):
<no Python frame>
```

I tried the installation multiple times with different `.bashrc` configs:
- without `export LD_LIBRARY_PATH`
- with `export LD_LIBRARY_PATH=/usr/local/lib`
- with no `export PYTHONPATH`
- with `export PYTHONPATH=/usr/local/lib/python3.8/site-packages`
- with `export PYTHONPATH=/usr/local/lib/python3.9/site-packages` (where the vapoursynth .la and .so files were, even when I had my virtualenv enabled)
- with export `PYTHONPATH=/home/me/.pyenv/versions/encoding/lib/python3.9/site-packages` (softlinked the vapoursynth .la and .so files from python3.9/site-packages into here)

I'm not sure what I'm doing wrong here. Any help would be appreciated 😄

**Comment by cid-chan** [comment]
Can you try this:

`./autogen.sh && ./configure && make && sudo make install` installs vapoursynth system-wide.
Install vapoursynth with pip or your whatever pyenv offers within your venv, in addition to those three commands

**Comment by BirdExpert123** [comment]
Thank you for the help @cid-chan

I tried those commands out and they worked fine however I'm quite sure I have a problem on my end with my Python paths. Even after setting both `PYTHONHOME` and `PYTHONPATH` I seem to get the same error message. I tried a lot of variations out and even re-installed pyenv and went through the steps again, to no avail.

I'll be closing the issue as the problem isn't really vapoursynth related since I'm pretty sure it's something to do with pyenv/the virtualenv.
